core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -459,9 +461,9 @@ class LogSegment(val log: FileRecords,
    * Change the last modified time for this log segment
    */
   def lastModified_=(ms: Long) = {
-    log.file.setLastModified(ms)
-    index.file.setLastModified(ms)
-    timeIndex.file.setLastModified(ms)
+    Files.setLastModifiedTime(log.file.toPath(), FileTime.fromMillis(ms))
+    Files.setLastModifiedTime(index.file.toPath(), FileTime.fromMillis(ms))
+    Files.setLastModifiedTime(timeIndex.file.toPath(), FileTime.fromMillis(ms))
```

Comment:
Maybe we should reuse the same `FileTime`? Also a nit: `toPath()` should just be `toPath`, there are a few cases like that in the PR.